internal/flink/command_connection.go

```diff
@@ -0,0 +1,147 @@
+	cmd.Flags().String("aws-access-key", "", fmt.Sprintf("Specify access key for the type: %s.", utils.ArrayToCommaDelimitedString(flink.ConnectionSecretTypeMapping["aws-access-key"], "or")))
+	cmd.Flags().String("aws-secret-key", "", fmt.Sprintf("Specify secret key for the type: %s.", utils.ArrayToCommaDelimitedString(flink.ConnectionSecretTypeMapping["aws-secret-key"], "or")))
+	cmd.Flags().String("aws-session-token", "", fmt.Sprintf("Specify session token for the type: %s.", utils.ArrayToCommaDelimitedString(flink.ConnectionSecretTypeMapping["aws-session-token"], "or")))
+	cmd.Flags().String("service-key", "", fmt.Sprintf("Specify service key for the type: %s.", utils.ArrayToCommaDelimitedString(flink.ConnectionSecretTypeMapping["service-key"], "or")))
+	cmd.Flags().String("username", "", fmt.Sprintf("Specify username for the type: %s.", utils.ArrayToCommaDelimitedString(flink.ConnectionSecretTypeMapping["username"], "or")))
+	cmd.Flags().String("password", "", fmt.Sprintf("Specify password for the type: %s.", utils.ArrayToCommaDelimitedString(flink.ConnectionSecretTypeMapping["password"], "or")))
+}
+
+func validateConnectionType(connectionType string) error {
+	if !slices.Contains(flink.ConnectionTypes, connectionType) {
+		return errors.NewErrorWithSuggestions("invalid connection type "+connectionType, fmt.Sprintf("Specify the connection type as %s.", utils.ArrayToCommaDelimitedString(flink.ConnectionTypes, "or")))
+	}
+	return nil
+}
+
+func validateConnectionSecrets(cmd *cobra.Command, connectionType string) (map[string]string, error) {
+	var connectionSecrets []string
+	connectionSecrets = append(connectionSecrets, flink.ConnectionTypeSecretMapping[connectionType]...)
+
+	for k := range flink.ConnectionSecretTypeMapping {
+		secret, err := cmd.Flags().GetString(k)
+		if err != nil {
+			return nil, err
+		}
+		if secret != "" && !slices.Contains(connectionSecrets, k) {
+			return nil, errors.NewErrorWithSuggestions(fmt.Sprintf("%s is invalid for connection %s.", k, connectionType), fmt.Sprintf("Valid secret types are %s.", utils.ArrayToCommaDelimitedString(connectionSecrets, "or")))
+		}
+	}
+
+	requiredSecretKeys := flink.ConnectionRequiredSecretMapping[connectionType]
+	var optionalSecretKeys []string
+	for _, secretKey := range flink.ConnectionTypeSecretMapping[connectionType] {
+		if !slices.Contains(requiredSecretKeys, secretKey) {
+			optionalSecretKeys = append(optionalSecretKeys, secretKey)
+		}
+	}
+
+	secretMap := map[string]string{}
+	for _, requiredKey := range requiredSecretKeys {
+		secret, err := cmd.Flags().GetString(requiredKey)
+		if err != nil {
+			return nil, err
+		}
+		if secret == "" {
+			return nil, fmt.Errorf("must provide %s for type %s", requiredKey, connectionType)
+		}
+		backendKey, ok := flink.ConnectionSecretBackendKeyMapping[requiredKey]
+		if !ok {
+			return nil, fmt.Errorf("backend key not found for %s", requiredKey)
```

Comment:
This is a good practice to avoid confusion in case `requiredKey` is `""`.
```suggestion
			return nil, fmt.Errorf(`backend key not found for "%s"`, requiredKey)
```